bundles/specmate-migration-test/src/com/specmate/migration/test/AddAttributeTest.java

```diff
@@ -0,0 +1,229 @@
+package com.specmate.migration.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.event.EventAdmin;
+import org.osgi.service.log.LogService;
+import org.osgi.util.tracker.ServiceTracker;
+
+import com.specmate.common.SpecmateException;
+import com.specmate.migration.api.IMigratorService;
+import com.specmate.migration.test.attributeadded.testmodel.artefact.ArtefactFactory;
+import com.specmate.migration.test.attributeadded.testmodel.artefact.Diagram;
+import com.specmate.migration.test.attributeadded.testmodel.base.Folder;
+import com.specmate.migration.test.baseline.testmodel.base.BaseFactory;
+import com.specmate.migration.test.support.AttributeAddedModelProviderImpl;
+import com.specmate.migration.test.support.BaselineModelProviderImpl;
+import com.specmate.migration.test.support.ServiceController;
+import com.specmate.model.support.util.SpecmateEcoreUtil;
+import com.specmate.persistency.IPackageProvider;
+import com.specmate.persistency.IPersistencyService;
+import com.specmate.persistency.ITransaction;
+import com.specmate.persistency.cdo.internal.CDOPersistencyService;
+import com.specmate.persistency.cdo.internal.config.CDOPersistenceConfig;
+import com.specmate.urihandler.IURIFactory;
+
+public class AddAttributeTest {
```

Comment:
General remark: To test addition of an attribute, maybe using the id attribute is not the best idea, as several other services rely on this attribute being present. In fact executing the test triggers several exceptions that are written on the console. My suggestion would be to use a new attribute.

Reply:
I'm not sure I'm following. Do you refer to the fact that the ecore model objects have an attribute "id"? The tests on this level are intended to be agnostic of the business logic, so the attribute could be called "foo" or "bar", for the tested functionality (adding an attribute) it does not matter.
It is due to the lack of my imagination that the ecore model has these object and attributes. I can rename them and model, for example, food, to make clear that the models that are tested have nothing in common with the models used in specmate.

Reply:
yes, the tests are agnostic about the attribute but, e.g. the persistency service is not. For instance it generates events when something changes, and for these events it uses the id attribute. Hence, when you run the current tests, you will find errros on the console "could not determine uri from model" or something like this. This is due to the missing id attribute.

Reply:
Ah, I see. Ok.